File: front-end/src/app/models/fundraiser.ts
import { Donation } from "./donation";
import { Photo } from "./photo";

export interface Fundraiser {
    id: number;
    name: string;
    description: string;
    createdAt: Date;
    currentAmount: number;
    targetAmount: number;
    isValidated: boolean;
    isRejected:boolean;
    url: string;
    userId: number;
    photos: Photo[];
}

File: API/Entities/Fundraiser.cs
using System;
using System.Collections.Generic;
using HelpAFamilyOfferAChance.API.Entities;

namespace API.Entities
{
    public class Fundraiser
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public double CurrentAmount { get; set; }
        public double TargetAmount { get; set; }
        public bool IsValidated { get; set; }
        public bool IsRejected { get; set; }
        public User User { get; set; }
        public int UserId { get; set; }
        
        public ICollection<Photo> Photos { get; set; }
        public ICollection<Donation> Donations { get; set; }
        public ICollection<FavouriteList> FavouriteList { get; set; }


    }
}